src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,307 @@
+    protected $connection;
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $finSearchConfigRepository;
+
+    /**
+     * @var array[]
+     */
+    private $configs = [];
+
+    public function __construct(EntityRepositoryInterface $finSearchConfigRepository, Connection $connection)
+    {
+        $this->finSearchConfigRepository = $finSearchConfigRepository;
+        $this->connection = $connection;
+    }
+
+    /**
+     * @return array|bool|float|int|string|null
+     */
+    public function get(string $key, string $salesChannelId, string $languageId)
+    {
+        $config = $this->load($salesChannelId, $languageId);
+
+        $parts = explode('.', $key);
+
+        $pointer = $config;
+
+        foreach ($parts as $part) {
+            if (!\is_array($pointer)) {
+                return null;
+            }
+
+            if (\array_key_exists($part, $pointer)) {
+                $pointer = $pointer[$part];
+
+                continue;
+            }
+
+            return null;
+        }
+
+        return $pointer;
+    }
+
+    public function getString(string $key, string $salesChannelId, string $languageId): string
+    {
+        $value = $this->get($key, $salesChannelId, $languageId);
+        if (!\is_array($value)) {
+            return (string)$value;
+        }
+
+        throw new InvalidSettingValueException($key, 'string', \gettype($value));
+    }
+
+    public function getInt(string $key, string $salesChannelId, string $languageId): int
+    {
+        $value = $this->get($key, $salesChannelId, $languageId);
+        if (!\is_array($value)) {
+            return (int)$value;
+        }
+
+        throw new InvalidSettingValueException($key, 'int', \gettype($value));
+    }
+
+    public function getFloat(string $key, string $salesChannelId, string $languageId): float
+    {
+        $value = $this->get($key, $salesChannelId, $languageId);
+        if (!\is_array($value)) {
+            return (float)$value;
+        }
+
+        throw new InvalidSettingValueException($key, 'float', \gettype($value));
+    }
+
+    public function getBool(string $key, string $salesChannelId, string $languageId): bool
+    {
+        return (bool)$this->get($key, $salesChannelId, $languageId);
+    }
+
+    public function all(string $salesChannelId, string $languageId): array
+    {
+        return $this->load($salesChannelId, $languageId);
+    }
+
+    /**
+     * @throws InvalidDomainException
+     * @throws InvalidUuidException
+     * @throws InconsistentCriteriaIdsException
+     */
+    public function getDomain(string $salesChannelId, string $languageId): array
+    {
+        $domain = 'FinSearch';
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->select('LOWER(HEX(id))')
+            ->from('finsearch_config')
+            ->where('sales_channel_id = :salesChannelId')
+            ->andWhere('language_id = :languageId');
+
+        $domain = rtrim($domain, '.') . '.';
```

Comment:
You are declaring `$domain` in line `122` as `FinSearch`. All this trimming and str_replace doesn't even do anything, so why are we doing it in the first place?

The only thing that is being done, is adding a `.` to the end. Why isn't this a constant then?